From daa34fcc44f05e61316665c0837b79153f1c0ab4 Mon Sep 17 00:00:00 2001
From: "Guilherme G. Piccoli" <gpiccoli@igalia.com>
Date: Wed, 22 Nov 2023 16:44:58 -0300
Subject: [PATCH 068/188] PM: suspend: Also disable s2idle on Galileo (OLED
 Deck)

Reports are that s2idle might also be bogus on Galileo - since S3/deep
is tested and working, let's include the new HW in the s2idle ban.

Signed-off-by: Guilherme G. Piccoli <gpiccoli@igalia.com>
(cherry picked from commit f65259f588545c8e811a3ca4e34c2a172969dbd7)
---
 kernel/power/suspend.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/power/suspend.c b/kernel/power/suspend.c
index 0911ad540..7ccec8425 100644
--- a/kernel/power/suspend.c
+++ b/kernel/power/suspend.c
@@ -209,7 +209,8 @@ void __init pm_states_init(void)
 	product_name = dmi_get_system_info(DMI_PRODUCT_NAME);
 
 	if (sys_vendor && (!strncmp("Valve", sys_vendor, 5)) &&
-	    product_name && (!strncmp("Jupiter", product_name, 7))) {
+	    ((product_name && (!strncmp("Jupiter", product_name, 7))) ||
+	    (product_name && (!strncmp("Galileo", product_name, 7))))) {
 		s2idle_unsupported = true;
 		pr_info("Steam Deck quirk - no s2idle allowed!\n");
 	}
-- 
2.55.0

